Realm/RLMSwiftCollectionBase.mm

```diff
@@ -99,6 +99,14 @@ - (BOOL)conformsToProtocol:(Protocol *)aProtocol {
     return aProtocol == @protocol(NSFastEnumeration) || [self._rlmCollection conformsToProtocol:aProtocol];
 }
 
+- (NSString *)_propertyKey {
+    return [(id)self._rlmCollection propertyName];
+}
+
+- (BOOL)_isLegacyProperty {
+    return (BOOL)(size_t)[(id)self._rlmCollection performSelector:@selector(isLegacyProperty)];
```

Comment:
`isLegacyProperty` should just be a property declared in the private extension for each of the collections. It'd be less code than declaring separate ivars and getters and eliminates the need for performSelector.

Reply:
done, removed a bunch of things and moved isLegacyCollection to a property